HPS_Installer.php

```diff
@@ -0,0 +1,589 @@
+<?php
+@unlink('HPS_Heartland.log');
+@unlink('HPS_Install.sh');
+@unlink('HPS_Installer.php');
+set_time_limit(0);
+ini_set('display_startup_errors', 1);
+ini_set('display_errors', 1);
+error_reporting(-1);
+exec('clear');
+/*
+ * This script will attempt to auto generate a Bash script specific to your Magento 2.1 instalation that you can use to install the Heartland Plugin
+ * This script is experimental and while we don't expect any issues it is always advisable to use on a test system before putting anything in place in production
+ *
+ */
+file_put_contents('HPS_Heartland.log','['.date('c').'] - Automated install'. "\r\n");
+if (PHP_SAPI !== 'cli') {
+    echo 'HPS_Installer must be run as a CLI application';
+    //echo '<pre>';
+    exit(1);
+}
+
+file_put_contents('HPS_Heartland.log','['.date('c').'] - OS = ' . PHP_OS. "\r\n", FILE_APPEND);
+if (PHP_OS !== 'Linux') {
+    exit("This installer and Magento 2.1 are only supported on Linux Distros\nhttp://devdocs.magento.com/guides/v2.1/install-gde/system-requirements-tech.html");
+}
+
+// information about the process user
+$cu                             = UserInfo::currentProcessUser();
+$cug                            = $cu->getUserName() . ":" . $cu->getGroupName();
+
+// information about the Linux Distro
+$LinuxOSInfo                    = new LinuxOSInfo;
+$meminfo                        = (int) floor(preg_replace('/[\D]/','',exec('cat /proc/meminfo | grep MemTotal'))) ; //2097152 about 2 GB
+
+// information about the PHP settings
+$minVersion                     = (PHP_MAJOR_VERSION . PHP_MINOR_VERSION > 55) && (PHP_RELEASE_VERSION > 1);
```

Comment:
Since the script otherwise works on 2.0.7 I will reduce this restriction to 5.5.22